beta-src/src/utils/state/gameApiSlice/extraReducers/fetchGameData/precomputeLegalOrders.ts

```diff
@@ -0,0 +1,524 @@
+import provincesMapData from "../../../../../data/map/ProvincesMapData";
+import TerritoryMap from "../../../../../data/map/variants/classic/TerritoryMap";
+import Province from "../../../../../enums/map/variants/classic/Province";
+import Territory from "../../../../../enums/map/variants/classic/Territory";
+import { UnitType } from "../../../../../models/enums";
+import { IProvinceStatus, ITerritory } from "../../../../../models/Interfaces";
+import { GameData } from "../../../../../state/interfaces/GameDataResponse";
+import GameOverviewResponse from "../../../../../state/interfaces/GameOverviewResponse";
+import { GameState } from "../../../../../state/interfaces/GameState";
+import GameStateMaps from "../../../../../state/interfaces/GameStateMaps";
+
+interface LegalVia {
+  dest: Territory;
```

Comment:
Should you put `src` here as well? A comment would be nice explaining what `provIDPaths` is and whether it includes the src and dest.

Is this *all* possible paths? Why do you need this? Couldn't this get exponentially large?

Reply:
I'll add a comment. We need at least the set of all minimal paths (minimal under the partial ordering where you check if one path is a subsequence of another), because when you then try to compute legal supports, if a fleet is the one doing the support, you need to be able to tell if there is still a convoy path that does *not* use that fleet, and you want to compute all the possible supports by any fleet, you need to be able to answer that question for any fleet.

But computing the full set of paths is much easier than computing the set of minimal paths, so that's what we do.

Yes it's exponential. But I took some time to eyeball the brachiness of the seas on the map, and as far as I can tell it's only like O(100) at worst, and that's when all the seas are filled and you are looking for vias that go from Turkey all the way up to England/Scandanavia.

Hopefully that's not too bad. We can look at other options if this is slow, there is a console.log that measures the legal moves computation time.

Reply:
Oh interesting, I didn't think about that. I appreciate the explanation. (copying this explanation into a comment might be helpful for the next reader)